python/tvm/relay/op/random/kernel.py

```diff
@@ -132,3 +132,52 @@ def foo(key):
         :py:func:`threefry_generate`.
     """
     return _make.threefry_split(key)
+
+
+def uniform(key, shape, dtype="float32", low=0.0, high=1.0):
+    """Draw samples from a uniform distribution.
+
+    Samples are uniformly distributed over the half-open interval [low, high)
+    (includes low, but excludes high). In other words, any value within the
+    given interval is equally likely to be drawn by uniform.
+
+    Example
+    -------
+
+    .. code-block:: python
+
+        key = threefry_key(0)
+        random_values = uniform(key, (100,), low=0, high=10)
+
+    Parameters
+    ----------
+    key : relay.Expr
+        key that uniquely determines the random values. Multiple uses with the
+        same generator will generate the same random values. This generator should be
+        treated as an opaque pointer. You can create one from calling
+        :py:func:`threefry_key`, :py:func:`threefry_split`, or
+        :py:func:`threefry_generate`. **Do not use this generator again after calling
+        this function.**
+
+    shape : Sequence[int]
+        Desired outputs shape of random numbers.
+
+    dtype : str
+        Desired outputs type of random numbers.
+
+    low : float or relay.Expr, optional
+        Lower bound of the uniform distribution.
+
+    high : float or relay.Expr, optional
+        Upper bound of the uniform distribution.
+
+    Returns
```

Comment:
One thing we want to think about here is the API for generating random values. The main question is around handling the threefry key. Do we want the user to split the key before every invocation, or do we want anything that uses the key to return the next one.

i.e.

```
key_1, key_2 = split(key)
data = uniform(key_2, ...)
my_other_random_funtion(key_1)
```

vs

```
data, key = uniform(key, ...)
my_other_random_function(key)
```

Personally I prefer the second. It also makes better use of the key space and avoid recomputing the key too often.

Reply:
Agree~ I've updated the output.